drivers/generic/cpssEnabler/linuxNoKernelModule/drivers/ethDriver.c

```diff
@@ -3688,12 +3703,22 @@ int mvppnd_create_netdev(struct mvppnd_dev *ppdev, const char *name, int port)
 	       flow->config_tx_dsa_size);
 
 	if (!flow_id) {
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5,15,0)
 		memcpy(ndev->dev_addr, DEFAULT_MAC, ETH_ALEN);
+#else
+		dev_addr_mod(ndev, 0, DEFAULT_MAC, ETH_ALEN);
+#endif
 	} else {
 		/* MAC is based on port 0 MAC with last byte set to flow ID */
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5,15,0)
 		memcpy(ndev->dev_addr, ppdev->sdev.flows[0]->ndev->dev_addr,
 		       ETH_ALEN);
 		ndev->dev_addr[5] = flow_id;
+#else
+		dev_addr_mod(ndev, 0, ppdev->sdev.flows[0]->ndev->dev_addr,
+		             ETH_ALEN);
+		dev_addr_mod(ndev, 5, &flow_id, 1);
```

Comment:
dev_addr_mod() internally calls memcpy(), and flow_id is long. This incidentally works for little endian targets, but for compatibility with big endian targets, you need either to retain the original code which assigns directly, or first copy flow_id to a u8 type, and pass that u8 variable to dev_addr_mod().

Reply:
Thanks Elad, Will change the code to first copy flow_id to a u8 type, and pass that u8 variable to dev_addr_mod().

for direct assignment and memcpy, now dev_addr is changed to "const unsigned char *"  because of the direct assignment is not getting allowed by compiler and it throws compilation error like below. similar case for memcpy for dev_addr. because of that community kernel code is also changed like this.   

/sonic/platform/marvell-arm64/prestera/mrvl-prestera/drivers/generic/cpssEnabler/linuxNoKernelModule/drivers/ethDriver.c:3569:35: error: assignment of read-only location '*(ndev->dev_addr + 5)'
 3569 |                 ndev->dev_addr[5] = flow_id;

/sonic/platform/marvell-arm64/prestera/mrvl-prestera/drivers/generic/cpssEnabler/linuxNoKernelModule/drivers/ethDriver.c:3567:28: 
note: expected 'void *' but argument is of type 'const unsigned char *'
 3567 |                 memcpy(ndev->dev_addr, ppdev->sdev.flows[0]->ndev->dev_addr,
      |                        ~~~~^~~~~~~~~~